vm/src/org/mozartoz/truffle/nodes/call/CallMethodNode.java

```diff
@@ -33,11 +45,52 @@ public static CallMethodNode create() {
 
 	static final RecordFactory OTHERWISE_MESSAGE_FACTORY = Arity.build("otherwise", 1L).createFactory();
 
-	@Specialization
+	@Specialization(guards = {
+			"self.getClazz() == cachedClazz",
+			"cachedLabel.equals(labelFromArgs(args))",
```

Comment:
All literals (which are the values which can appear as a label) can be compared by identity, so you can use `==` here, which is particularly important for Atom/java.lang.String (we don't want byte by byte comparison).